lib/cloud_controller/dependency_locator.rb

```diff
@@ -434,14 +413,23 @@ def build_opi_instances_client
     end
 
     def build_bbs_instances_client
-      bbs_client = ::Diego::Client.new(
+      VCAP::CloudController::Diego::BbsInstancesClient.new(build_bbs_client)
+    end
+
+    def build_bbs_client
+      connect_timeout = config.get(:diego, :bbs, :connect_timeout)
+      send_timeout = config.get(:diego, :bbs, :send_timeout)
+      receive_timeout = config.get(:diego, :bbs, :receive_timeout)
+
+      ::Diego::Client.new(
         url: config.get(:diego, :bbs, :url),
         ca_cert_file: config.get(:diego, :bbs, :ca_file),
         client_cert_file: config.get(:diego, :bbs, :cert_file),
         client_key_file: config.get(:diego, :bbs, :key_file),
+        connect_timeout: connect_timeout.nil? ? 10 : connect_timeout,
```

Comment:
For new configurable properties like this we are defaulting them at the BOSH job spec level (like this [UAA timeout](https://github.com/cloudfoundry/capi-release/blob/972a14749c928de1283908023deac75f6708fcfc/jobs/cloud_controller_ng/spec#L655-L657) for example) since this gives CF platform operators a single place to look for defaults. I'd prefer for this to be defaulted there and this code could just read directly from the config.

Reply:
The reason, why I provided this here is, so that the capi-release PR does not have to be synced with the CC sub-module bump. To avoid breaking the build, I could submit a follow-up PR which takes the default away here. Or do you accept a breakage, or a synced PR+sub-module bump?

Reply:
Hmm that's a good point, but I don't think this particular set of changes should break the build if we merge the https://github.com/cloudfoundry/capi-release/pull/110/ PR first. In general though, our usual protocol is to merge them at roughly the same time and this usually results in no build breakages due to how our pipelines are structured.

I think it's simpler overall to just omit the defaults in this PR and have the pair that merges this take https://github.com/cloudfoundry/capi-release/pull/110 in to consideration.

Reply:
Okay. Will update the PR soon.